app/src/main/java/com/example/android/moviedbtrainingapp/view/MainActivity.java

```diff
@@ -1,213 +0,0 @@
-        }
-        return super.onOptionsItemSelected(item);
-    }
-
-    public void downloadMoviesData() {
-        MovieCallback callback = MovieRestManager.getClient().create(MovieCallback.class);
-        Observable<MoviesResponse> movies = callback.getMovies(moviesOrder, Constants.API_KEY.API_KEY, page);
-        movies.subscribeOn(Schedulers.newThread())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(new Observer<MoviesResponse>() {
-                    @Override
-                    public void onCompleted() {
-                        swipeRefreshLayout.setRefreshing(false);
-                    }
-
-                    @Override
-                    public void onError(Throwable e) {
-                        loadingFailure();
-                        swipeRefreshLayout.setRefreshing(false);
-                    }
-
-                    @Override
-                    public void onNext(MoviesResponse moviesResponse) {
-                        if (moviesOrder.equals(Constants.MOVIES_ORDER.TOP_MOVIES)) {
-                            List<Movie> moviesList = moviesResponse.getResults();
-                            topRatedMovies.addAll(moviesList);
-                            endlessList(topRatedMovies);
-                            topRatedPage++;
-                        }
-                        if (moviesOrder.equals(Constants.MOVIES_ORDER.POPULAR_MOVIES)) {
-                            List<Movie> moviesList = moviesResponse.getResults();
-                            popularMovies.addAll(moviesList);
-                            endlessList(popularMovies);
-                            popularPage++;
-                        }
-                        loading = false;
-                        page++;
-                    }
-                });
-    }
-
-    public void endlessList(List<Movie> movies) {
-        movieRecycle.addOnScrollListener(new RecyclerView.OnScrollListener() {
-            @Override
-            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
-                if (dy > 0) {
-                    visibleItemCount = movieLinearManager.getChildCount();
-                    totalItemCount = movieLinearManager.getItemCount();
-                    pastVisibleItems = movieLinearManager.findFirstVisibleItemPosition();
-
-                    if (!loading) {
-                        if ((visibleItemCount + pastVisibleItems) >= totalItemCount) {
-                            spinner.setVisibility(View.VISIBLE);
-                            loading = true;
-                            downloadMoviesData();
-                        }
-                    }
-                }
-            }
-        });
-        adapter.getMovies(movies);
-        adapter.notifyDataSetChanged();
-        spinner.setVisibility(View.GONE);
-    }
-
-    public void config() {
-        movieLinearManager = new LinearLayoutManager(this);
-        adapter = new MovieAdapter(this, this);
```

Comment:
Passing the same object twice into your own method isn't good idea. Change constructor so it will wait for one object and do with it what you want inside.